src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1614,18 +1614,17 @@ private void HandleDirective(DirectiveDescriptor descriptor)
                         break;
 
                     case DirectiveTokenKind.String:
-                        if (At(CSharpSymbolType.StringLiteral))
+                        if (At(CSharpSymbolType.StringLiteral) && CurrentSymbol.Errors.Count == 0)
```

Comment:
The parser would classify incomplete string literals `"aksfj` as string literals with errors. Added this error check to ensure we drop down into our own error case and not consume string literals that aren't valid.